File: frontend/lib/api/status.ts
import { StatusResponse, UpdateStatusRequest } from "@/types/roomStatus";
import { API_URL, authHeaders, unwrap } from ".";

export const getPublicStatus = async (username: string) => {
  const res = await fetch(`${API_URL}/status/${username}`);
  return unwrap<StatusResponse>(res);
};

export const getMyStatus = async (token: string) => {
  const res = await fetch(`${API_URL}/status/me`, {
    method: "GET",
    headers: authHeaders(token),
  });
  return unwrap<StatusResponse>(res);
};

export const updateStatus = async (token: string, req: UpdateStatusRequest) => {
  const res = await fetch(`${API_URL}/status/me`, {
    method: "PUT",
    headers: authHeaders(token),
    body: JSON.stringify({
      preset_id: req.presetId,
      custom_message: req.customMessage,
    }),
  });
  return unwrap<StatusResponse>(res);
};

File: backend/internal/model/status.go
package model

import "time"

type RoomStatus struct {
	ID            string    `json:"id"`
	UserID        string    `json:"user_id"`
	PresetID      string    `json:"preset_id"`
	CustomMessage string    `json:"custom_message"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type StatusUpdateRequest struct {
	PresetID      string `json:"preset_id"`
	CustomMessage string `json:"custom_message"`
}

type StatusResponse struct {
	DisplayName   string `json:"display_name"`
	Preset        Preset `json:"preset"`
	CustomMessage string `json:"custom_message"`
}
